src/plot_api/plot_api.js

```diff
@@ -1726,6 +1794,9 @@ exports.relayout = function relayout(gd, astr, val) {
 
     if(Object.keys(aobj).length) gd.changed = true;
 
+    cleanDeprecatedLayoutAttributes(aobj);
+    cleanDeprecatedLayoutAttributeStrings(aobj);
```

Comment:
Could we call these two things from `_relayout` to avoid duplication?

Reply:
Absolutely! Done.